module-gui/gui/Common.hpp

```diff
@@ -79,60 +79,72 @@ namespace gui
         Horizontal
     };
 
-    template <class T>
-    bool operator&(const T &lhs, const T &rhs)
+    // This namespace exists to prevent instantiation of operators for other classes from namespace gui
+    namespace rectangle_enums
     {
-        return static_cast<uint32_t>(lhs) & static_cast<uint32_t>(rhs);
-    }
-    template <class T>
-    T operator|(const T &lhs, const T &rhs)
-    {
-        return static_cast<T>(static_cast<uint32_t>(lhs) | static_cast<uint32_t>(rhs));
-    }
-    template <class T>
-    T operator|=(const T &lhs, const T &rhs)
-    {
-        lhs = lhs | rhs;
-        return lhs;
-    }
-
-    enum class RectangleEdge
-    {
-        None   = 0x00,
-        Top    = 0x01,
-        Bottom = 0x02,
-        Left   = 0x04,
-        Right  = 0x08,
-        All    = 0x0F
-    };
-
-    enum class RectangleRoundedCorner
-    {
-        None        = 0x00,
-        TopLeft     = 0x10,
-        TopRight    = 0x20,
-        BottomLeft  = 0x40,
-        BottomRight = 0x80,
-        All         = 0xF0,
-    };
-
-    enum class RectangleFlatEdge
-    {
-        None        = 0x00,
-        TopLeft     = 0x01,
-        TopRight    = 0x02,
-        BottomLeft  = 0x04,
-        BottomRight = 0x08
-    };
+        enum class RectangleEdge
+        {
+            None   = 0x00,
+            Top    = 0x01,
+            Bottom = 0x02,
+            Left   = 0x04,
+            Right  = 0x08,
+            All    = 0x0F
+        };
+
+        enum class RectangleFlatEdge
+        {
+            None        = 0x00,
+            TopLeft     = 0x01,
+            TopRight    = 0x02,
+            BottomLeft  = 0x04,
+            BottomRight = 0x08
+        };
+
+        enum class RectangleRoundedCorner
+        {
+            None        = 0x00,
+            TopLeft     = 0x10,
+            TopRight    = 0x20,
+            BottomLeft  = 0x40,
+            BottomRight = 0x80,
+            All         = 0xF0,
+        };
+
+        enum class RectangleYap
+        {
+            None        = 0x00,
+            TopLeft     = 0x10,
+            TopRight    = 0x20,
+            BottomLeft  = 0x40,
+            BottomRight = 0x80,
+        };
+
+        // If the above scoped enums require operators then maybe they should be regular enums?
```

Comment:
Let's resolve this issue instead of leaving comment ;)

Reply:
I mean this solution works and limits the scope of these operators. My point was that if we use these enums like integers performing bitwise operations and implicitly casting them to `bool` then maybe these shouldn't be class enums but regular enums instead? Since they are used as such.

Reply:
I removed the comment and one operator that was not used anywhere. I think this is still better than it was, i.e. the scope of the operators is limited, and can stay as it is for now.